File: src/main/java/indi/dkx/laocai/bot/model/entity/FriendEntity.java
package indi.dkx.laocai.bot.model.entity;

import indi.dkx.laocai.bot.model.enums.Sex;

/**
 * 好友实体
 * @param userId 用户 QQ 号
 * @param nickname 用户昵称
 * @param sex 用户性别，可能值：male female unknown
 * @param qid 用户 QID
 * @param remark 好友备注
 * @param category 好友分组
 */
public record FriendEntity(
        Long userId,
        String nickname,
        Sex sex,
        String qid,
        String remark,
        FriendCategoryEntity category
) { }

File: laocai-bot-spring-boot-starter/src/main/kotlin/indi/kyson/laocai/bot/entity/FriendEntity.kt
package indi.kyson.laocai.bot.entity

import com.fasterxml.jackson.annotation.JsonProperty
import indi.kyson.laocai.bot.enums.Sex

data class FriendEntity(
    @JsonProperty("user_id")
    val userId: Long,
    val nickname: String,
    val sex: Sex,
    val qid: String,
    val remark: String,
    val category: FriendCategoryEntity,
)
